### PR TITLE
Fix document fields readonly

### PR DESCRIPTION
**Pull request by Lainow**
## Checklist before requesting a review

*Please delete options that are not relevant.*

- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] This change requires a documentation update.

## Description

- It fixes !34821
- Here is a brief description of what this PR does

The document fields could not be passed as readonly, the problem arose when a user wanted to pass a Fields field as readonly but this failed. And the field could still be modified.

## Screenshots (if appropriate):
image

### CODE REVIEW

**Review by Rom1-B — APPROVED** [review]

File: src/Document.php [hunk 1]
```diff
@@ -1714,11 +1714,21 @@ public static function dropdown($options = [])
             $values[$data['id']] = $data['name'];
         }
         $rand = mt_rand();
-        $out  = Dropdown::showFromArray('_rubdoc', $values, ['width'               => '30%',
+        $readonly = $p['readonly'] ?? false;
```
Comment (cedric-anne) [inline src/Document.php:1717]: Please add a `$p['readonly'] = false` on line 1673. Having all the default options values at the same place is easier to read.

```suggestion
        $readonly = $p['readonly'];
```

**Review by cedric-anne — COMMENTED** [review]
Seems OK.

**Review by trasher — APPROVED** [review]
OK for me after change @cedric-anne requested

**Comment by trasher** [comment]
You must declare readonly in `$p` array at the beginning of the method

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**